homeassistant/components/elmo_alarm/__init__.py

```diff
@@ -0,0 +1,265 @@
+from collections import namedtuple
+from datetime import timedelta
+import logging
+
+from elmo.api.client import ElmoClient
+from elmo.api.exceptions import PermissionDenied
+from urllib3.exceptions import HTTPError
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_CUSTOM_BYPASS,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_call_later
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "elmo_alarm"
+
+CONF_VENDOR = "vendor"
+
+CONF_STATES = "states"
+CONF_ZONES = "zones"
+
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=30)
+DEFAULT_HOST = "https://connect.elmospa.com"
+
+SIGNAL_ZONE_CHANGED = f"{DOMAIN}.zone_changed"
+SIGNAL_INPUT_CHANGED = f"{DOMAIN}.input_changed"
+SIGNAL_ARMING_STATE_CHANGED = f"{DOMAIN}.arming_state_changed"
+
+ZONE_ARMED = 0
+ZONE_DISARMED = 1
+
+INPUT_ALERT = 1
+INPUT_WAIT = 0
+
+ZoneData = namedtuple("ZoneData", ["zone_id", "zone_name", "state"])
+InputData = namedtuple("InputData", ["input_id", "input_name", "state"])
+
+
+STATE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): vol.In(
+            [
+                STATE_ALARM_ARMED_HOME,
+                STATE_ALARM_ARMED_AWAY,
+                STATE_ALARM_ARMED_NIGHT,
+                STATE_ALARM_ARMED_CUSTOM_BYPASS,
+            ]
+        ),
+        vol.Required(CONF_ZONES): vol.All(cv.ensure_list),
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_HOST, default=DEFAULT_HOST): cv.string,
+                vol.Required(CONF_VENDOR): cv.string,
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): vol.All(cv.time_period, cv.positive_timedelta),
+                vol.Required(CONF_STATES): vol.All(cv.ensure_list, [STATE_SCHEMA]),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the e-connect Elmo Alarm platform."""
+
+    conf = config[DOMAIN]
+
+    host = conf[CONF_HOST]
+    vendor = conf[CONF_VENDOR]
+    username = conf[CONF_USERNAME]
+    password = conf[CONF_PASSWORD]
+    states = conf[CONF_STATES]
+    scan_interval = conf[CONF_SCAN_INTERVAL]
+
+    client = ElmoClientWrapper(host, vendor, username, password, states)
+    await client.update()
+
+    hass.data[DOMAIN] = client
+
+    hass.async_create_task(
+        async_load_platform(
+            hass,
+            "binary_sensor",
```

Comment:
Please import `DOMAIN` from [.../binary_sensor/init.py](https://github.com/home-assistant/home-assistant/blob/e019280d94261688795a5d32c9a1a7a5e7948836/homeassistant/components/binary_sensor/__init__.py#L18) and use it here:
```suggestion
            DOMAIN,
```

Reply:
Done.